### PR TITLE
Issue #7746: Update AbstractChecks to log DetailAST - MissingJavadocP…

### PR DESCRIPTION
**Pull request by gaurabdg**
Fixes Issue #7746 

Updates:
https://github.com/checkstyle/checkstyle/blob/9b94d177d01b1d2c71f79b01a66359caa2f874b8/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocPackageCheck.java#L95

Diff Report: https://gaurabdg.github.io/checkstyle-tester-reports/regression/MissingJavadocPackage/diff/

### CODE REVIEW

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionMissingJavadocPackageTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    private final String checkName = MissingJavadocPackageCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testAnnotation() throws Exception {
```
Comment (baratali) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionMissingJavadocPackageTest.java]: why name is `testAnnotation`? better name would be `testBlockComment`.
Or like others do - `testOne`, `testTwo`, ...

**Comment by baratali** [comment]
diff report looks correct

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionMissingJavadocPackageTest.java (hunk 1 above)
Comment (gaurabdg) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionMissingJavadocPackageTest.java]: Totally my bad, I changed the logic, forgot to change the method name. Will do it asap

Comment (gaurabdg) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionMissingJavadocPackageTest.java]: @baratali Done, please review

**Comment by baratali** [comment]
@gaurabdg , I'm ok with the changes. However, checks failed.
```
[15:42:02][Step 2/2] The build Checkstyle::IDEA Inspections Pull Request #14342 {buildId=2823545} has been running for more than 50 minutes. Terminating...
```
Can you please re-push your branch to trigger checks again?

**Comment by gaurabdg** [comment]
@baratali Done

**Pull request merged**